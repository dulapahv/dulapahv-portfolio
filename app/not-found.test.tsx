import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import NotFound from './not-found';

describe('Not Found Page', () => {
  it('should render page not found heading', () => {
    render(<NotFound />);
    expect(screen.getByRole('heading', { name: 'Page Not Found' })).toBeInTheDocument();
  });

  it('should render return home link', () => {
    render(<NotFound />);
    const homeLink = screen.getByRole('link', { name: /return home/i });
    expect(homeLink).toBeInTheDocument();
    expect(homeLink).toHaveProperty('href', 'http://localhost:3000/');
  });

  it('should render error message', () => {
    render(<NotFound />);
    expect(
      screen.getByText(/sorry, the page you are looking for does not exist/i)
    ).toBeInTheDocument();
  });

  it('should display 404 status code', () => {
    render(<NotFound />);
    expect(screen.getByText('Status: 404')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<NotFound />);
    expect(container).toMatchSnapshot();
  });
});
