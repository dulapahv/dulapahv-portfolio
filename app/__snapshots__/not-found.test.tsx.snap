// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Not Found Page > should match snapshot 1`] = `
<div>
  <main
    class="space-y-4"
  >
    <header>
      <h1
        class="text-3xl font-semibold"
      >
        Page Not Found
      </h1>
    </header>
    <p>
      Sorry, the page you are looking for does not exist. Please check the URL and try again.
    </p>
    <a
      class="bg-mirai-red flex w-fit cursor-pointer items-center justify-center gap-2 rounded-md px-3 py-2 text-sm font-medium text-white select-none hover:bg-mirai-red/90 hover:!transition-colors! hover:shadow-md active:scale-[0.98] active:transition-transform! disabled:cursor-not-allowed disabled:opacity-50 disabled:hover:shadow-none"
      href="/"
    >
      <svg
        class="size-4.5 shrink-0"
        fill="currentColor"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M219.31,108.68l-80-80a16,16,0,0,0-22.62,0l-80,80A15.87,15.87,0,0,0,32,120v96a8,8,0,0,0,8,8h64a8,8,0,0,0,8-8V160h32v56a8,8,0,0,0,8,8h64a8,8,0,0,0,8-8V120A15.87,15.87,0,0,0,219.31,108.68ZM208,208H160V152a8,8,0,0,0-8-8H104a8,8,0,0,0-8,8v56H48V120l80-80,80,80Z"
        />
      </svg>
      Return Home
    </a>
  </main>
  <footer
    class="text-foreground-muted mt-8 border-t border-gray-200 pt-6 dark:border-gray-800"
  >
    <p
      class="mb-2 font-semibold"
    >
      Details:
    </p>
    <code
      class="block text-sm break-all whitespace-pre-wrap sm:text-base"
    >
      Status: 404
    </code>
  </footer>
  <footer
    class="mt-16"
  >
    <div
      class="text-foreground-muted flex flex-col gap-y-1 text-sm font-medium"
    >
      <p>
        Â©
         
        2007
         
        Dulapah Vibulsanti. All rights reserved.
      </p>
      <p>
        View the
         
        <a
          class="text-mirai-red underline underline-offset-2 hover:decoration-2"
          href="https://github.com/dulapahv/dulapahv-portfolio"
          rel="noopener noreferrer"
          target="_blank"
        >
          source code
        </a>
        .
      </p>
      <p>
        NLWeb Chat:
         
        <a
          class="text-mirai-red underline underline-offset-2 hover:decoration-2"
          href="https://chat.dulapahv.dev"
          rel="noopener noreferrer"
          target="_blank"
        >
          chat.dulapahv.dev
        </a>
        .
      </p>
    </div>
  </footer>
</div>
`;
